src/events/start-blockchain-event-listeners.js

```diff
@@ -1,41 +1,83 @@
 "use strict";
 
-var async = require("async");
-var addFilter = require("./add-filter");
-var subscriptions = require("./subscriptions");
+var compact = require("lodash").compact;
 var contracts = require("../contracts");
 var ethrpc = require("../rpc-interface");
 var isFunction = require("../utils/is-function");
 var isObject = require("../utils/is-object");
 var noop = require("../utils/noop");
+var parseLogMessage = require("./parse-message/parse-log-message");
+
+function parseLogs(logs, eventByAddressAndSignature) {
+  const parsedLogs = logs.map((log) => {
+    if (eventByAddressAndSignature[log.address] === undefined) {
+      throw new Error(`Missing mapping for contract address ${log.address}`);
+    }
+    const eventDetails = eventByAddressAndSignature[log.address][log.topics[0]];
+    // Based on how we grab event logs, it's possible to get an event we don't care about if
+    // two contracts have same event, but we only want to subscribe to one of them
+    if (eventDetails === undefined) return null;
+    return parseLogMessage(eventDetails.contractName, eventDetails.eventName, log, eventDetails.abi);
+  });
+  return compact(parsedLogs);
+}
+
+// Iterates through eventCallbacks, generates eth filter and event signature mapping
+function getEventsDetails(eventsToSubscribe, activeContracts) {
+  var eventByAddressAndSignature = {};
+  const addresses = [];
+  const topics = new Set();
+  for (const contractName in eventsToSubscribe) {
+    if (Object.prototype.hasOwnProperty.call(eventsToSubscribe, contractName)) {
```

Comment:
Perhaps `if (!Object.prototype.hasOwnProperty.call(eventsToSubscribe, contractName)) { continue; }` to lower cyclomatic complexity of inner loop

Reply:
We use eslint rule `no-continue`